specs/trip_dispatch_spec.rb

```diff
@@ -111,4 +92,95 @@
       expect(passenger.trips).must_include trip
     end
   end
+
+  describe "Request trip method" do
+    it "returns an instance of trip" do
+      new_trip = @dispatcher.request_trip(3)
+
+      expect(new_trip).must_be_kind_of RideShare::Trip
+    end
+
+    it "updates the passenger trip list" do
+      current_passenger = @dispatcher.find_passenger(3)
+      current_number_of_trips = current_passenger.trips.length
+      @dispatcher.request_trip(3)
+
+      expect(current_passenger.trips.length).must_equal (current_number_of_trips + 1)
+    end
+
+    it "updates the driver trip list" do
+      current_driver = @dispatcher.find_driver(5)
+      current_number_of_trips = current_driver.driven_trips.length
+      @dispatcher.request_trip(3)
+
+      expect(current_driver.driven_trips.length).must_equal (current_number_of_trips + 1)
+    end
+
+    it "ensures that driver assigned is available" do
+      driver = @dispatcher.check_driver_availability_and_assign(3)
+
+      expect(driver.status).must_equal :AVAILABLE
+    end
+
+    it "raises an ArgumentError if there are no available drivers" do
+      @dispatcher.request_trip(6)
+      @dispatcher.request_trip(3)
+
+      expect{@dispatcher.request_trip(1)}.must_raise ArgumentError
+    end
+
+    it "checks that drivers are not driving themselves" do
```

Comment:
This is a good test to do, but I don't think it tests what you want it to test. You should test it in a circumstance when the ONLY available driver is the passenger that you are trying to drive.